### ISSUE TITLE
Idea about flux management

### ISSUE THREAD

**Issue by Cenotaffio** [Azanor/thaumcraft-suggestions#321]
Flux is essentially a impure an out-of-control vis.
Actually there are no way to deal with it.
Could be a way to eliminate flux be to use it for magic?
obviously is ridiculous think that is possible to use flux for actually useful magic.
But, what about use the rift as catalyst to summon crimson creatures?
I mean, make the rift spend flux in order to summon monsters, then kill them so that nothing remain.
Also totem-like magic could be used to maintain large areas clear forcing all the flux in a large area into the "controlled" rift to make the rift use it for summoning creatures that then can be killed.
Also the flux could be spent by the rift to produce dungeons-like structures inside them that if not destroyed could cause invasions.
Caution have to be made, because if for some reasons too much flux is feeded in the rift, or the summon process is not keep under control (maybe because the essential stabilization machines run out of vis, or get damaged, or destroyed), the rift can summon really dangerous monsters (as dangerous as the wither), or worst collapse with a explosion and a great flux production.

**Comment by Reassembly** [comment]
You have some good ideas, but you're starting with the assumption that there isn't _already_ a way to deal with flux, or any ways to make productive use of it.

What you are describing seems like a combination of some of the methods for dealing with flux that already exist in TC6, and the Outer Lands dimension/dungeon that was in TC4/5, but has not been added to TC6 yet.

**Comment by praecipitator** [comment]
There is no way of dealing with flux, see https://github.com/Azanor/thaumcraft-beta/issues/922#issuecomment-394286769

Otherwise, I'm not sure about summoning, but I like the idea about controlling a rift, instead of closing it. Build something like the Node Stabilizers from TH4 above and below it, for example. It would then allow flux to leave via the rift, but prevent taint from entering. As long as you don't mess up, of course.

And yes, the rift could then be used to enter the Far Lands, instead of having to look for obelisks. "Hey, there's that tear in the space-time continuum where all that weird contamination is coming from! Let's jump right into it!"

**Comment by Cenotaffio** [comment]
I know that my idea wasn't that original.
But i think was worth of suggestion.
Also, the fact that the suggestions already resemble some parts of the mod, made me think that this could make them more feasible.

Made flux disappear without cost seem quite boring to me, so i don't think that an flux sponge like item, would be a good idea, neither a machine that simply make the same work of a sponge (with some process in the middle) seem appealing.
In my opinion flux should be a serious problem at every stage of the game, and the only safe way of treating it should be to avoid the production.
Make little amount of flux manageable is a must, but made it safe at some point of the game should be avoided.
Maybe make a some sort of flux trap, that don't destroy flux but contain it, and if the trap is broken or not managed correctly, bad things happens.
maybe some sort of barrier that isolate the flux in the boundaries of a specific chunk, then send the unwanted flux in that chunk that act like a radiation deposit facility.
In the boundaries of that chunk taint start to spread and generate monsters, if the flux is too high for the barrier, or if monsters manage to surpass the defenses and broke the barrier....you know what could happen.
(make a flux containing facility with golem and turrets is my dream)

**Comment by Cenotaffio** [comment]
sorry for eventual bad grammar

**Comment by praecipitator** [comment]
Flux deposit facility. That might be interesting :D

- Flux traps sucks flux from the local chunk (or maybe 9 chunks with that vis relay upgrade), and stores it into adjacent flux batteries.
- Flux batteries can store a certain amount of flux. However, they can be overfilled beyond the safe capacity. The further beyond a battery is, the higher the probability of catastrophic failure. If picked up, the battery will also release all the stored flux. However, they can be put on a minecart for transportation. The block version should output a comparator signal representing the fill status.
- The Flux Containment Unit is basically just an extra large battery. It's a multiblock structure which isn't mobile in any way. Similar to the battery, it can suffer catastrophic failure if overfilled and outputs a comparator signal.
- The Flux Pump is basically a hopper for flux, to transfer flux between batteries, minecarts with batteries and containment units. A Vent can also be attached to it's output, enabling it to output flux back into the atmosphere. Redstone can be used to turn the pump on or off

For obvious reasons, you can't have any kind of flux storage items.

**Comment by praecipitator** [comment]
As an additional hazard, the flux containment unit could automatically chunkload and attrackt mobs which try to destroy it. Then you would actually need defense there :D

Also, as a long-term solution, there could be a late-game way to actually turn the contained flux into an item, albeit very, very slowly. So that if you are running an infernal furnace and are smelting essentia on a daily basis, there is no way you can process the flux as fast as you are generating it.

**Comment by Cenotaffio** [comment]
Also, flux batteries and containment units has to use energy or essentia in order to function, so that you have to actually build a containment facility where put those batteries, if a great amount of flux is put in the air in a short time period, it could cause the explosion of the batteries, or cause a peak in the required energy that they use to function, so a breach in a single battery could cause a chain reaction.

I don't know what are the Azanor's ideas about flux, but in my opinion flux is the other side of the progress, like warp, you can contain it's effects, but if you want to be a great mage, you have to make sacrifices, like curses that spread in you mind and in the world around you.
Many people want a way to get rid of flux, but i think that is more in the TC style something similar to "you can't destroy Flux, but you can seal it and prey that nothing terrible happen" than "Flux is just a way to push you to the game stage were it is no more a problem".
(similarly to factorio's pollution)

Also a result of the flux could be the "rebellion" of your magic Systems, like golem that get corrupted or infusion altars that start abominable rituals.
For the transport with minecarts a better solution could be that normal batteries can't be put on cart, but you can made an apposite special cart, that have the problem to be unstable, and while full of flux it start spawning mob that can cause damages to the railroad or even destroy the cart itself.

(barriers are a must for the containing of flux)

**Comment by praecipitator** [comment]
Flux ~~battery~~ capacitor!
Capacitors can be charged and discharged quickly, but don't have much capacity and can only hold charge for a short time. And they can breakdown if overcharged.

Let's say 50 points of flux can be safely stored in a capacitor, and a pump stores 1 point per tick.
But for each second, the capacitor has a small chance to release 1 point back. The more it contains, the higher the probability. Also, each time 1 point of taint is added above the 50, there is a probability that the capacitor explodes. The more charge it contains, the higher the probability.
And yes, exploding capacitors should definitely cause a chain reaction :D

Also yes, a special cart might be better. It has a somewhat lower capacity, but doesn't leak as much. However, it is also less stable. It will explode if the cart stops too sudden (like when it collide with something or derails), or if it goes around a corner too fast. Then going around corners somewhat fast or accelerating/decelerating it somewhat too roughly then could do "minor" things, like spawning mobs.

The containment unit would then be the permanent storage solution. How about this: It has pretty much infinite capacity, but it runs on vis, and the more flux it contains, the more vis it requires. Explain it like: the higher the internal flux pressure, the stronger you have to push against it.

If the vis levels drops below what the containment unit requires, bad stuff happens. If it only contains a little flux, like let's say 200, it will just start leaking. But when it's more, like 600+, the unit will violently explode, contaminating a large area and creating a rift at the very least. Maybe even something worse than just a rift. Remember hungry nodes from TC4? Imagine a rift which draws everything into itself, and converts it into even more flux.

To troll players even more, the containment unit could attract cultists who don't directly attack it, but actively try to drain the vis in the area.

**Comment by Reassembly** [comment]
TL;DR: Planted essentia crystals absorb flux. That much is _already_ implemented. Maybe build on that?

@katzenpapst  The comment you reference is talking about _rifts,_ which there is (currently) no way to _remove_ once they open. _Flux_ is connected to rifts, but it is _a different thing_ - and controlling or exploiting something is also not the same as removing it.

From the top:
- **Rift Control:** Stabilizers powered by Vis Generators pointed at a rift will not _remove_ the rift, but will stabilize it and eventually entirely stop it from spawning _any_ taint entities.
- **Flux Control:** Essentia crystals planted on stone _will absorb flux_ from the chunk they're in, turning into Vitium crystals - 10 points of flux per crystal. They can then be mined out and recycled into Salis Mundus for making fresh crystals. If the crystals are planted before a rift opens, and are replaced when they get full of flux, the flux cannot build up enough to open a rift. Even after a rift opens, soaking up flux from the area with crystals may reduce the severity of taint outbreaks. 
- **Profit:** The taint seeds spawned by an unstabilized rift will convert nearby stone/cobblestone into hard-to-break tainted rock. After the taint seed is killed, the tainted rock will slowly decay into easy-to-break porous stone. Porous stone is an _incredibly_ rich source of random essentia crystals, native clusters, amber, quartz, diamonds, emeralds, and other minerals - especially if mined with a tool/focus with a few levels of Fortune enchantment. (Also, the process of tainting the stone uses up some flux - so you're basically converting flux into resources.)

All of this is _already_ implemented, and already works. The only thing that's significantly missing is a way to remove rifts, which is a planned addition.

@everyone_else  As I said above, planted essentia crystals basically _are_ flux capacitors, in that they absorb and safely store flux from their environment. I still like the idea of an _actual_ "Flux Capacitor", which could be a more efficient (but maybe riskier) replacement for planted crystals in the later stages of research, much like how the Smelter/Thaumatorium are (messy but convenient) upgrades to the Crucible. It would certainly help out anyone who gets that far without noticing that planted essentia crystals absorb flux.

Did I mention that planted essentia crystals absorb flux? _Planted essentia crystals absorb flux._

**Comment by praecipitator** [comment]
Well, I never had the need to actually grow any crystals, they are abundant as-is. Did the text change somewhen between upgrades, too? I usually don't re-read entries unless the changelogs mention a change to that specific reserach.

And hm. Could it be that I have never broken porous stone with a pickaxe? I usually clean up the aftermaths of taint with my exchange/silktouch focus, so I always assumed porous stone is just useless. I have a few stacks of them by now, I think.

**Comment by Cenotaffio** [comment]
flux is actually quite manageable, if you are aware of the danger and focus your effort to limit the production while using crystals and the natural absorption that every chunk has, is actually quite difficult to have taint and rifts, (the only time that i got a rift is when i actually voluntarily caused it).
Also taint is manageable, seeds spawn only to ground level, and until there are a LOT of flux or more than one seed, is really easy to dispose of them (arrows or fire/lava).
Moreover the spread of the taint is actually quite slow, the problem is only when you are on a server with a chunkloader and taint spread while you are offline.
I wanted to suggest more fun ways to see flux, cause in my opinion is almost too easy for experienced players, i don't want a super hardcore flux system, but a management that grows in the late game, and that is not obliterated with the reach of certain level.
